cmd/ws/listener.go

```diff
@@ -74,11 +75,34 @@ func (s *Listener) ListenAndServe() error {
 
 	router.HandleFunc("/", s.Connect)
 
-	go func() {
-		if err := s.server.ListenAndServe(); err != nil && err != http.ErrServerClosed {
-			log.Fatal("ListenAndServe error: " + err.Error())
+	if useTLS {
+		config := &tls.Config{
+			// Only allow ciphers that support forward secrecy for iOS9 compatibility:
+			// https://developer.apple.com/library/prerelease/ios/technotes/App-Transport-Security-Technote/
+			CipherSuites: []uint16{
+				tls.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
+				tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+				tls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
+				tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+			},
+			PreferServerCipherSuites: true,
+			MinVersion:               tls.VersionTLS12,
 		}
-	}()
+
+		s.server.TLSConfig = config
+
+		go func() {
+			if err := s.server.ListenAndServeTLS("./cert/cert.pem", "./cert/key.pem"); err != nil && err != http.ErrServerClosed {
```

Comment:
Either use the certs in the clientfakes dir or make the client tests use these certs